src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(allTodos => {
+      setIsLoading(false);
+      setTodos(allTodos);
+    });
+  }, []);
+
+  const reset = useCallback(() => {
+    setQuery('');
+  }, []);
+
+  const showModal = useCallback(
+    (todo: Todo) => {
+      setSelectedTodo(todo);
+    },
+    [],
+  );
+
+  const hideModal = () => {
+    setSelectedTodo(null);
+  };
+
+  const visibleTodos = todos.filter(todo => {
+    const isInclude = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    switch (selectFilter) {
+      case 'all':
+        return isInclude;
+
+      case 'active':
+        return !todo.completed && isInclude;
+
+      case 'completed':
+        return todo.completed && isInclude;
+
+      default:
+        throw new Error('error!');
+    }
+  });
```

Comment:
let's create separate function that will transform todos and return the visible ones